agenta-backend/agenta_backend/tasks/evaluations.py

```diff
@@ -112,45 +128,24 @@ def evaluate(
 
             # 3. We evaluate
             evaluators_results: [EvaluationScenarioResult] = []
-            for evaluator_config_id in evaluation.evaluators_configs:
-                evaluator_config = loop.run_until_complete(
-                    fetch_evaluator_config(evaluator_config_id)
-                )
-
-                additional_kwargs = (
-                    {
-                        "app_params": app_variant_parameters,
-                        "inputs": data_point,
-                    }
-                    if evaluator_config.evaluator_key == "auto_custom_code_run"
-                    else {}
-                )
+            for evaluator_config_db in evaluator_config_dbs:
                 result = evaluators_service.evaluate(
-                    evaluator_config.evaluator_key,
-                    app_output.output,
-                    data_point["correct_answer"],
-                    evaluator_config.settings_values,
-                    **additional_kwargs,
+                    evaluator_key=evaluator_config_db.evaluator_key,
+                    variant_output=app_output.output,
+                    correct_answer=data_point["correct_answer"],
+                    settings_values=evaluator_config_db.settings_values,
+                    app_params=app_variant_parameters,
+                    inputs=data_point,
                 )
 
                 result_object = EvaluationScenarioResult(
-                    evaluator_config=evaluator_config.id,
+                    evaluator_config=evaluator_config_db.id,
                     result=result,
                 )
                 evaluators_results.append(result_object)
-                if (
-                    evaluators_aggregated_data[evaluator_config_id]["evaluator_key"]
-                    == ""
-                ):
-                    evaluators_aggregated_data[evaluator_config_id][
-                        "evaluator_key"
-                    ] = evaluator_config.evaluator_key
-                evaluators_aggregated_data[evaluator_config_id]["results"].append(
-                    result
-                )
 
-            # 4. We create a new evaluation scenario
-            evaluation_scenario = loop.run_until_complete(
+            # 4. We save the result of the eval scenario in the db
```

Comment:
here we create a new evaluation scenario, saving the results is just after
```
    loop.run_until_complete(
        update_evaluation_with_aggregated_results(
            new_evaluation_db.id, aggregated_results
        )
```

Reply:
I have checked. The function create_new_evaluation_scenario saves an evaluation scenario to the db as mentioned in the comment. Update evaluation with aggregated results adds the aggregated results for the whole eval